packages-exp/functions-compat/src/service.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  httpsCallable as httpsCallableExp,
+  useFunctionsEmulator as useFunctionsEmulatorExp,
+  getFunctions
+} from '@firebase/functions-exp';
+import { FirebaseFunctions, HttpsCallable } from '@firebase/functions-types';
+import { HttpsCallableOptions, Functions } from '@firebase/functions-types-exp';
+import { FirebaseApp } from '@firebase/app-types';
+import { FirebaseError } from '@firebase/util';
+
+export class FunctionsService implements FirebaseFunctions {
+  private _functionsInstance: Functions;
+  /**
+   * For testing.
+   * @internal
+   */
+  _region: string;
+  /**
+   * For testing.
+   * @internal
+   */
+  _customDomain: string | null;
+
+  constructor(public app: FirebaseApp, regionOrCustomDomain?: string) {
+    this._functionsInstance = getFunctions(app, regionOrCustomDomain);
```

Comment:
Can you get the functions-exp instance from the component framework directly using the string name `functions-exp` in `register.ts` and pass it into the constructor?

What you have here is not wrong, but it will be easier to see dependencies between components this way. It  will also make testing easier by allowing you to stub the functions-exp instance easily.

Reply:
Done.